### PR TITLE
Fixed the counting problem

### PR DESCRIPTION
**Pull request by ArrogantGao**
A few fix:
1. in previous version, the reducer and results type is not properly delivered, I think in `reduce_and_branch`, `reducer` and `results_type` should not be key word arguments. (I only fixed the error but did not changed the interface)
2. `MISCount` now is only used by `mis1` `mis2` and `xiao2013`, I defined `MaxSize` and `MaxSizeBranchCount` in Core and changed the algebra in `reduce_and_branch`.
3. I change the `apply_branch` function to return resulting problem only, and add a `apply_branch_gain` to give both the problem and the gain.

### CODE REVIEW

File: lib/OptimalBranchingCore/src/interfaces.jl [hunk 1]
```diff
@@ -16,12 +16,17 @@ Create a branch from the given clause applied to the specified vertices.
 - `vertices`: A vector of vertices to be considered for the branch.
 
 ### Returns
-A tuple of two values:
 - `AbstractProblem`: A new instance of `AbstractProblem` with reduced size.
-- `Number`: The local gain of the branch, which will be added to the global gain.
 """
 function apply_branch end
 
+"""
```
Comment (GiggleLiu) [inline lib/OptimalBranchingCore/src/interfaces.jl:23]: How about still use `apply_branch`, but returns an object that contains both the result and the gain.

The interface not computing the gain seems to be not very useful.

**Review by GiggleLiu — COMMENTED** [review]
The rest looks very good!

**Comment by GiggleLiu** [comment]
fix #8
fix #7

File: lib/OptimalBranchingCore/src/interfaces.jl (hunk 1 above)
Comment (ArrogantGao) [reply lib/OptimalBranchingCore/src/interfaces.jl:23]: The function `optimal_branching_rule` use `apply_branch` to see the difference of measure, personally I do not want to send `result_type` into this function.

Comment (GiggleLiu) [reply lib/OptimalBranchingCore/src/interfaces.jl:23]: I think the following code is also intuitive.
```julia
measure(apply_branch(problem, candidate.clause, variables).problem, m)
```

**Comment by ArrogantGao** [comment]
I removed the key words agruements in `reduce_and_bound()`, and I revomed the `apply_branch_gain()`.

**Comment by ArrogantGao** [comment]
User interfaces added. I added two functions `mis_size` and `mis_branch_count`. One can simply input a `AbstractGraph` into them and get the results, without the need to set anything manually.

**Comment by ArrogantGao** [comment]
Tests has been added to enhance the codecov rate

File: lib/OptimalBranchingCore/src/interfaces.jl [hunk 2]
```diff
@@ -6,7 +6,7 @@ The problem type that can be used in the optimal branching framework.
 abstract type AbstractProblem end
 
 """
-    apply_branch(problem::AbstractProblem, clause::Clause, vertices::Vector)::Tuple{AbstractProblem, Number}
+    apply_branch(problem::AbstractProblem, clause::Clause, vertices::Vector)::Tuple{AbstractProblem}
```
Comment (GiggleLiu) [inline lib/OptimalBranchingCore/src/interfaces.jl:9]:
```suggestion
    apply_branch(problem::AbstractProblem, clause::Clause, vertices::Vector)::Tuple
```

**Review by GiggleLiu — APPROVED** [review]

**Pull request merged**